### PR TITLE
Make sure that the chef indexer finished (bsc#988961)

### PR DESCRIPTION
**Pull request by rsalevsky**
After deploying the crowbar role, solr takes some time to index the new
role. This change force rebuilds the index to prevent empty searches.

### CODE REVIEW

**Comment by jdsn** [comment]
:+1:

**Comment by toabctl** [comment]
+1 (assuming it was tested and works).

**Comment by Itxaka** [comment]
Seems to work ok locally. I noticed that this only works in chef<11 so maybe we should set a note in case we move from chef10 in the future? from chef>=11 we need to use `chef-server-ctl reindex` instead

+1

**Comment by vuntz** [comment]
Asking out loud: would the code from https://github.com/rsalevsky/crowbar/blob/b9da555dfa69766ff89fdb73c7afe85c5e756052/scripts/install-chef-suse.sh#L1037 help?

**Comment by vmoravec** [comment]
`knife index rebuild -y` turns out not to be working for the successful first script run, I succeeded only after putting `sleep 60` on the next line. 
-1

**Comment by rsalevsky** [comment]
@vuntz We can try it out... Do you think we should move the code up our should we have the check twice?

**Comment by vmoravec** [comment]
@vunt @rsalevsky Moving https://github.com/rsalevsky/crowbar/blob/b9da555dfa69766ff89fdb73c7afe85c5e756052/scripts/install-chef-suse.sh#L1037..L1046 up below the `wait_for_crowbar` line helped, yay!

**Comment by vuntz** [comment]
> @vuntz We can try it out... Do you think we should move the code up our should we have the check twice?

I'd probably do it twice, just to be safe.

**Comment by vmoravec** [comment]
@vuntz Second time it always seems to return only a single line: `Expander Queue Total =        0` , the first time the output is more verbose:
```ruby 
Expander Queue Total =        16
Expander Queue Total =        16        
Expander Queue Total =        15    
Expander Queue Total =        15    
Expander Queue Total =        4    
Expander Queue Total =        4    
Expander Queue Total =        2    
Expander Queue Total =        0    
```

**Comment by rsalevsky** [comment]
I think it doesn't hurt us at the moment to check it twice. Long term we should get anyway rid of this check.

PS: I added the patch to the D:C:7:S package in case someone want's to test it.

File: scripts/install-chef-suse.sh [hunk 1]
```diff
@@ -323,6 +323,19 @@ wait_for_crowbar ()
     fi
 }
 
+wait_for_chef ()
+{
+    local count=0
+    local value=10000
+    while (($count < 60 && $value !=0))
```
Comment (vuntz) [inline scripts/install-chef-suse.sh]: You might want to add a space before `0`. But minor detail.

**Comment by vuntz** [comment]
+1, thanks for the fix!

**Comment by aplanas** [comment]
+1

**Comment by toabctl** [comment]
+1

**Comment by rsalevsky** [comment]
Just added the missing space  before `0`.

**Pull request merged**